packages/auth/src/model/public_types.ts

```diff
@@ -291,6 +291,11 @@ export interface Auth {
     error?: ErrorFn,
     completed?: CompleteFn
   ): Unsubscribe;
+  /**
+   * return a promise that resolves immediately when the initial
+   * auth state is settled and currentUser is available
+   */
+  authStateReady(): Promise<void>;
```

Comment:
This will also resolve if a user signs out.. (since that will cause onAuthStateChanged to trigger as well). Let's add more more line to the comment like:

"the current user might be a valid user, or null if there is no user signed in currently."